uint8 num
DesiredSpeed[] speed
---
bool success